core/invoker/src/main/scala/whisk/core/containerpool/docker/ProcessRunner.scala

```diff
@@ -52,17 +52,75 @@ trait ProcessRunner {
         case _                 => None
       }
 
-      (process.exitValue(), out.mkString("\n"), err.mkString("\n"), scheduled)
+      (ExitStatus(process.exitValue()), out.mkString("\n"), err.mkString("\n"), scheduled)
     }).flatMap {
-      case (0, stdout, _, scheduled) =>
+      case (ExitStatus(0), stdout, _, scheduled) =>
         scheduled.foreach(_.cancel())
         Future.successful(stdout)
-      case (code, stdout, stderr, scheduled) =>
+      case (exitStatus, stdout, stderr, scheduled) =>
         scheduled.foreach(_.cancel())
-        Future.failed(ProcessRunningException(code, stdout, stderr))
+        timeout match {
+          case t: FiniteDuration if exitStatus.terminatedBySIGTERM =>
+            Future.failed(ProcessTimeoutException(timeout, exitStatus, stdout, stderr))
+          case _ => Future.failed(ProcessRunningException(exitStatus, stdout, stderr))
+        }
     }
+}
+
+case class ExitStatus(statusValue: Int) {
+
+  // Based on The Open Group Base Specifications Issue 7, 2018 edition:
+  // Shell & Utilities - Shell Command Language - 2.8.2 Exit Status for Commands
+  // http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_08_02
+  val STATUS_SUCCESSFULL = 0
```

Comment:
Better to move constants to `object` instance

Reply:
Done. Why did you suggest the change - just for style / coding convention reasons or are there additional reasons?

Reply:
Its a minor concern. In a case class each `val` would be transformed to a instance variable and add to memory footprint of `ExitStatus`. So it would be better to move them to `object` where it gets transformed to `static final` kind of variable and thus a single copy.

I do not expect too many `ExitStatus` instances and thus this memory overhead should not be a big concern. Still would be good to place such _constants_ in `object`.

Reply:
Thanks for this hint. I missed this part.